docs/flags.md

```diff
@@ -37,23 +37,44 @@ Here are the options flags can have:
-    multiple: false,              // allow setting this flag multiple times
-    env: 'MY_NAME',               // default to value of environment variable
-    options: ['a', 'b'],          // only allow the value to be from a discrete set
-    parse: input => 'output',     // instead of the user input, return a different value
-    default: 'world',             // default value if flag not passed (can be a function that returns a string or undefined)
-    required: false,              // make flag required (this is not common and you should probably use an argument instead)
-    dependsOn: ['extra-flag'],    // this flag requires another flag
-    exclusive: ['extra-flag'],    // this flag cannot be specified alongside this other flag
+    char: 'n',                                   // shorter flag version
+    summary: 'brief summary',                    // help summary for flag
+    helpGroup: 'THE BEST FLAGS',                 // Put flag into THE BEST FLAGS group in help
+    description: 'in-depth overview',            // help description for flag
+    hidden: false,                               // hide from help
+    multiple: false,                             // allow setting this flag multiple times
+    env: 'MY_NAME',                              // default to value of environment variable
+    options: ['a', 'b'],                         // only allow the value to be from a discrete set
+    parse: async input => 'output',              // instead of the user input, return a different value
+    default: 'world',                            // default value if flag not passed (can be a async function that returns a string or undefined)
+    required: false,                             // make flag required
+    dependsOn: ['extra-flag'],                   // this flag requires another flag
+    exclusive: ['extra-flag'],                   // this flag cannot be specified alongside this other flag
+    exactlyOne: ['extra-flag', 'another-flag'],  // exactly one of these flags must be provided
+    output
+    relationships: [                             // define complex relationships between flags
+      // Make this flag dependent on all of these flags
+      {type: 'all', flags: ['flag-one', 'flag-two']}
+      // Make this flag dependent on at least one of these flags
+      {type: 'some', flags: ['flag-three', 'flag-four']}
+      // Make this flag exclusive of all these flags
+      {type: 'none', flags: ['flag-five', 'flag-six']}
+
+      // Make this flag dependent on all of these flags
+      {type: 'all', flags: [
+        'flag-one',
+        'flag-two',
+        // Include flag-seven but only when flag-eight is equal to FooBar
+        {name: 'flag-seven', when: async (flags) => flags['flag-eight'] === 'FooBar'}
+      ]}
+    ]
   }),
 
   // flag with no value (-f, --force)
   force: Flags.boolean({
-    char: 'f',
+    char: 'f',                    // shorter flag version
```

Comment:
The word `version` gave me pause here but I'm not sure what would be better. Maybe `short flag name`?